libs/common/src/lib/config.ts

```diff
@@ -51,6 +51,7 @@ export const DEFAULT_PAGE_SIZE = 50;
 export const DEFAULT_PROCESSOR_CONCURRENCY_GATHER_ASSET_PROFILE = 1;
 export const DEFAULT_PROCESSOR_CONCURRENCY_GATHER_HISTORICAL_MARKET_DATA = 1;
 export const DEFAULT_PROCESSOR_CONCURRENCY_PORTFOLIO_SNAPSHOT = 1;
+export const DEFAULT_QUEUE_LOCK_DURATION_PORTFOLIO_SNAPSHOT = 30000;
```

Comment:
Maybe we should default this to something higher to unbreak the folks experiencing the crash without requiring them to set an extra variable? Normally I'd prefer using the library's default, but since the behavior prior to using Job queues was to keeping computing the portfolio snapshot until done without crashing, it might make sense to set this higher here.

Reply:
I am not yet sure whether this solution is appropriate, as `QUEUE_LOCK_DURATION` exposes a lot of implementation details.

Ideas I have in mind
* Expose something like a snapshot computation timeout
* Set the default to `0` or `30000` (as the library default)

Reply:
> I am not yet sure whether this solution is appropriate, as QUEUE_LOCK_DURATION exposes a lot of implementation details.

Good point, I agree.

> Expose something like a snapshot computation timeout

Sounds reasonable

> Set the default to `0` or `30000` (as the library default)

Do you know if `0` works? The [bull reference](https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#advanced-settings) doesn't mention it being a valid value. I tried to find where it's used, and I think it gets translated to the `NX` param to a Redis `SET` call [here](https://github.com/OptimalBits/bull/blob/c2f37ee194f0d16d78329e2fa4ea335db37350a6/lib/commands/takeLock-1.lua#L13), which seems to expect a positive value.

Reply:
> Do you know if `0` works? The [bull reference](https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#advanced-settings) doesn't mention it being a valid value.

Sorry, I mixed it up with [stalledInterval](https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queue):

```
interface AdvancedSettings {
  ...
  stalledInterval: number = 30000; // How often check for stalled jobs (use 0 for never checking).
  ...
}
```

My favorite at the moment: Name the variable `PROCESSOR_PORTFOLIO_SNAPSHOT_COMPUTATION_TIMEOUT` with a default value of `30000`. Users with issues can increase that number.